internal/stream_push_helper_test.go

```diff
@@ -167,3 +168,183 @@ func TestBackup_Retry_NetworkError(t *testing.T) {
 func TestSplitBackup_Retry_NetworkError(t *testing.T) {
 	checkPushAndFetchBackup(t, 3, 1009, 100*1000, 30*1000, 5, 1000*1000)
 }
+
+func GetS3Folder(networkErrorAfterByteSize int) (storage.Folder, func() error, error) {
+	cwd, err := filepath.Abs("./")
+	if err != nil {
+		return nil, nil, err
+	}
+	// Create temp directory.
+	tmpDir, err := os.MkdirTemp(cwd, "data")
+	if err != nil {
+		return nil, nil, err
+	}
+	err = os.Chmod(tmpDir, 0755)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	folder, err := fs.ConfigureFolder(tmpDir, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+	if networkErrorAfterByteSize != 0 {
+		return functests.NewS3ErrorFolder(folder, networkErrorAfterByteSize),
+			func() error {
+				return os.RemoveAll(tmpDir)
+			}, nil
+	} else {
+		return folder, func() error {
+			return os.RemoveAll(tmpDir)
+		}, nil
+	}
+}
+
+// We have to repeat testtools code to evade import cycle
```

Comment:
I don't like copy-pasting code...
I have seen following approach to solve circular dependencies - rename the package to `internal_test` (without moving file to new directory).
Not sure that it is best practice... however it works!